docs/reference/mapping/types/version.asciidoc

```diff
@@ -0,0 +1,125 @@
+[role="xpack"]
+[testenv="basic"]
+[discrete]
+[[version-field-type]]
+=== Version field type
+
+The `version` field type is a specialization of the `keyword` field for
+handling software version values and to support specialized precedence
+rules for them. Precedence is defined following the rules outlined by
+https://semver.org/[Semantic Versioning], which for example means that
+major, minor and patch version parts are sorted numerically (i.e. 
+"2.1.0" < "2.4.1" < "2.11.2") and pre-release versions are sorted before
+release versiond (i.e. "1.0.0-alpha" < "1.0.0").
+
+You index a `version` field as follows
+
+[source,console]
+--------------------------------------------------
+PUT my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "my_version": {
+        "type": "version"
+      }
+    }
+  }
+}
+
+--------------------------------------------------
+
+The field offers the same search capabilities as a regular keyword field. It 
+can e.g. be searched for exact matches using `match` or `term` queries and
+supports prefix and wildcard searches. The main benefit is that `range` queries
+will honour Semver ordering, so a `range` query between "1.0.0" and "1.5.0"
+will include versions of "1.2.3" but not "1.11.2" for example. Not that this
+would be different when using a regular `keyword` field for indexing where ordering
+is alphabetical.
+
+Software versions are expected to follow the
+https://semver.org/[Semantic Versioning rules] schema and precedence rules with
+the notable exception that more or less than three main version identifiers are
+allowed (i.e. "1.2" or "1.2.3.4" qualify as valid versions while they wouldn't under
+strict Semver rules). Version strings that are not valid under the Semver definition
+(e.g. "1.2.alpha.4") can still be indexed and retrieved as exact matches, however they
+will all appear _after_ any valid version with regular alphabetical ordering. 
```

Comment:
"with regular alphabetical ordering" =>  I am not quite sure what it means?  Are we sorting versions by an alphabetical ordering?

Reply:
Valid Semver versions like "1.2.3" are sorted numerically according to major, minor, patch, i.e. "1.2.1" < "1.11.1". There was a requirement to also be able to store and retrieve invalid versions like "1.a.123" for compatibility with existing solutions using "keyword", however we sort those after all valid versions and use the "default" keyword sorting which is alphabetical. Do the docs makes sense with this respect or do you think I can improve something?

Reply:
@cbuescher Thanks for explanation, with it, the docs completely make sense,  I somehow assumed that "alphabetical ordering" applied to valid versions.